services/setup/public/index.html

```diff
@@ -25,70 +34,102 @@ <h2>Services</h2>
-    const apps = await fetch('/apps').then(res => res.json());
+    const buildItemForm = buildFormElement(type);
 
-    apps.forEach(({ name, active }) => {
-      const label = document.createElement('label');
-      const span = document.createElement('span');
-      span.innerText = name;
+    elements
+      .map(buildItemForm)
+      .map((child) => selection.appendChild(child));
 
-      label.appendChild(span);
+    form.prepend(selection);
 
-      const checkbox = document.createElement('input');
-      checkbox.type = 'checkbox';
-      checkbox.name = 'apps[]';
-      checkbox.value = name;
-      checkbox.checked = active;
+    return form;
+  };
 
-      label.appendChild(checkbox);
+  const buildFormElement = (type) => ({ name, active }) => {
+    const id = `${type}-${name}`;
 
-      selection.appendChild(label);
-    });
+    const container = document.createElement('div');
+    container.className = 'input-group';
 
-    form.prepend(selection);
-  };
+    const label = document.createElement('label');
+    label.setAttribute('for', id);
 
-  const buildServices = async () => {
-    const form = document.getElementById('services-list');
-    const selection = document.createElement('div');
-    const services = await fetch('/services').then(res => res.json());
+    const span = document.createElement('span');
+    span.innerText = name;
+
+    label.appendChild(span);
 
-    services.forEach(({ name, active }) => {
-      const label = document.createElement('label');
-      const span = document.createElement('span');
-      span.innerText = name;
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.name = `${type}[]`;
+    checkbox.id = id;
+    checkbox.value = name;
+    checkbox.checked = active;
 
-      label.appendChild(span);
+    container.appendChild(checkbox);
+    container.appendChild(label);
 
-      const checkbox = document.createElement('input');
-      checkbox.type = 'checkbox';
-      checkbox.name = 'services[]';
-      checkbox.value = name;
-      checkbox.checked = active;
+    return container;
+  };
 
-      label.appendChild(checkbox);
+  const showNotification = (updateType) => {
+    const notifications = document.getElementsByTagName('mark');
 
-      selection.appendChild(label);
+    Array.prototype.forEach.call(notifications, (notification) => {
+      notification.id == updateType ?
+        notification.classList.remove('hidden') :
+        notification.classList.add('hidden');
     });
+  };
 
-    form.prepend(selection);
+  const buildControls = (publish) => {
+    document.getElementById('reload').onclick = () => {
+      publish('reload');
+      showNotification('reloaded');
+    };
+
+    document.getElementById('restart').onclick = () => {
+      publish('restart');
+      showNotification('restarted');
+    };
   };
 
   const app = async () => {
     const params = new URLSearchParams(document.location.search.substring(1));
```

Comment:
I know this is from a previous PR but I don't think the `substring` is necessary here. Passing `document.location.search` should just work.